lib/game.rb

```diff
@@ -0,0 +1,190 @@
+    #   return "Error! Do both players have the same token or did O go before X?"
+    # end
+  end
+
+  def win_array
+    WIN_COMBINATIONS.each do |win|
+      ref = self.board.cells[win[0]]
+      return [true, ref] if (ref == "X" || ref == "O") && self.board.cells[win[1]] == ref && self.board.cells[win[2]] == ref
+    end
+    [false, nil]
+  end
+
+  def won?
+    # WIN_COMBINATIONS.each do |win|
+    #   ref = self.board.cells[win[0]]
+    #   return true if (ref == "X" || ref == "O") && self.board.cells[win[1]] == ref && self.board.cells[win[2]] == ref
+    # end
+    # false
+    self.win_array[0]
+  end
+
+  def draw?
+    # return false if self.won?
+    # self.board.cells.each do |position|
+    #     return false if position == " "
+    # end
+    # true
+    return true if !self.won? && self.board.full?
+  end
+
+  def over?
+    return true if self.won? || self.draw?
+    false
+  end
+
+  def winner
+    #first implementation
+    # return nil if !self.won?
+    # o_count = 0
+    # x_count = 0
+    # self.board.cells.each do |position|
+    #   if position == "X"
+    #     x_count += 1
+    #   elsif position == "O"
+    #     o_count += 1
+    #   end
+    # end
+    # return "X" if x_count > o_count
+    # return "O" if o_count > x_count
+    # nil
+    # second implementation
+    # WIN_COMBINATIONS.each do |win|
+    #   ref = self.board.cells[win[0]]
+    #   return ref if (ref == "X" || ref == "O") && self.board.cells[win[1]] == ref && self.board.cells[win[2]] == ref
+    # end
+    # nil
+    self.win_array[1]
+  end
+
+  def player_turn_prompt(current_player, player_num)
+    valid = false
+    while !valid do
+      puts "Make your move, Player #{player_num}."
+      self.board.display
+      move = current_player.move(board)
+      if self.board.valid_move?(move)
+        self.board.update(move, current_player)
+        valid = true
+      else
+        puts "Invalid move. Instructions for playing Tic-Tac-Toe can be found here: https://en.wikipedia.org/wiki/Tic-tac-toe"
+      end
+    end
+  end
+
+  def turn
+    current_player = self.current_player # doing this assignment as an optimization as the current_player method requires a lot of computation
+    if current_player == self.player_1
+      # self.cli.player_turn_prompt(current_player, 1)
+      self.player_turn_prompt(current_player, 1)
+      # valid = false
+      # while !valid do
+      #   puts "Make your move, Player 1."
+      #   self.board.display
+      #   move = self.player_1.move(board)
+      #   if self.board.valid_move?(move)
+      #     self.board.update(move, player_1)
+      #     valid = true
+      #   else
+      #     puts "Invalid move. Instructions for playing Tic-Tac-Toe can be found here: https://en.wikipedia.org/wiki/Tic-tac-toe"
+      #   end
+      # end
+    elsif current_player == self.player_2
+      # self.cli.player_turn_prompt(current_player, 1)
+      self.player_turn_prompt(current_player, 1)
+      # valid = false
```

Comment:
I would really move out all of this old code. It makes it hard to read what is going on.